h2o-genmodel/src/main/java/hex/genmodel/algos/glm/GlmMojoModelBase.java

```diff
@@ -35,12 +49,112 @@ void init() { /* do nothing by default */ }
+      _column_mapping.put(_names[i], i);
+    }
+  }
+  
+  private void mapInteractions() {
+    if (_interaction_pairs == null) {
+      _interaction_mapping = null;
+      return;
+    }
+    
+    _interaction_mapping = new HashMap<String, Integer[]>();
+    for (StringPair pair: _interaction_pairs) {
+      int index1 = _column_mapping.get(pair.getFirst());
+      int index2 = _column_mapping.get(pair.getSecond());
+      _interaction_mapping.put((pair.getFirst() + "_" + pair.getSecond()), new Integer[]{index1, index2});
+    }
   }
 
+  /**
+   * Creates array that gives offset of each categorical-numerical interaction column, relative to the first
+   * categorical-numerical interaction column. First interaction column has offset 0, each offset is then
+   * determined by the domain length of the categorical variable in the subsequent interaction columns.
+   * Array is used to determine correct value of beta to use when scoring categorical-numerical interaction column
+   * and to impute the correct value into the categorical-numerical interaction columns when reading in data.
+   */
+  private void addCatNumOffsets() {
+    int catNumInteractions = 0;
+    for (int i = _cats; i < _domains.length - 1; i++) {
+      if (_domains[i] == null) {break;}
+      catNumInteractions++;
+    }
+    _catNumOffsets = new int[catNumInteractions + 1];
+    _catNumOffsets[0] = 0;
+    for (int i = 0; i < catNumInteractions; i++) {
+      _catNumOffsets[i + 1] = _catNumOffsets[i] + _domains[_cats + i].length;
+    }
+  }
+  
+  private boolean isInteraction(String name) {
+    if(_interaction_pairs == null) {return false;}
+    return _interaction_mapping.containsKey(name);
+  }
+  
+  private void imputeInteraction(double[] data, int index) {
+    Integer[] pair = _interaction_mapping.get(_names[index]);
+    // Impute categorical-categorical interaction
+    if (_domains[pair[0]] != null && _domains[pair[1]] != null) {
+      // If either categorical value in the pair is NaN, use interaction columns mode to impute
+      if(Double.isNaN(data[pair[0]]) || Double.isNaN(data[pair[1]])) {
+        data[index] = _catModes[index];
+      } else {
+        // Computes interaction value based on enum level of respective categorical values
+        data[index] = (_domains[pair[0]].length * data[pair[0]]) + data[pair[1]];  
+      }
+    }
+    // Impute numerical-numerical interaction
+    else if (_domains[pair[0]] == null && _domains[pair[1]] == null) {
+      if(Double.isNaN(data[pair[0]]) || Double.isNaN(data[pair[1]])) {
+        // meanOffset represents index at which categorical-numerical means end and numerical means begin in numMeans 
+        int meanOffset = _catNumOffsets[_catNumOffsets.length - 1];
+        // index - _cats - (_catNumOffsets.length - 1) says which numerical column is being used
+        data[index] = _numMeans[meanOffset + (index - _cats - (_catNumOffsets.length - 1))];
+      } else {
```

Comment:
here you can set an global 

_numMeanOffset = _catNumOffsets[_catNumOffsets.length - 1]-_cats - (_catNumOffsets.length - 1))]

then you can do 

data[index] = _numMeans[index+_numMeanOffset];